File: app/src/main/java/jp/ergo/android/screentouchdispatcher/ScreenTouchDispatcherBroadcastReceiver.java
package jp.ergo.android.screentouchdispatcher;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.preference.PreferenceManager;
import android.util.Log;
import android.widget.Toast;

public class ScreenTouchDispatcherBroadcastReceiver extends BroadcastReceiver {

    private static final String TAG = ScreenTouchDispatcherBroadcastReceiver.class.getSimpleName();

    public static final String ACTION_START_SERVICE = "jp.ergo.android.screentouchdispatcher.SATRT_SERVICE";
    public static final String ACTION_STOP_SERVICE = "jp.ergo.android.screentouchdispatcher.STOP_SERVICE";
    public static final String ACTION_SWITCH = "jp.ergo.android.screentouchdispatcher.SWITCH";

    @Override
    public void onReceive(Context context, Intent intent) {
        Log.d(TAG, "onReceive() " + intent.getAction());
        Toast.makeText(context, "受け取ったよ", Toast.LENGTH_LONG).show();

        final String action = intent.getAction();
        final Intent serviceIntent = new Intent(context, ScreenTouchDispatcherService.class);
        if (action.equals(ACTION_SWITCH)) {
            final boolean isDispatching = isDispatching(context);
            Log.d(TAG, "isDispatching: " + isDispatching);
            setDispatching(context, !isDispatching);
            if (isDispatching) {
                context.stopService(serviceIntent);
                NotificationUtils.sendNotification(context, "タッチして画面をロック");
            } else {
                context.startService(serviceIntent);
                NotificationUtils.sendNotification(context, "タッチしてロックを解除");

            }
        }
    }


    private boolean isDispatching(final Context context) {
        return PreferenceManager.getDefaultSharedPreferences(context).getBoolean("isDispatching", false);
    }

    private void setDispatching(final Context context, final boolean isDispatching) {
        PreferenceManager.getDefaultSharedPreferences(context).edit().putBoolean("isDispatching", isDispatching).commit();

    }

}

File: app/src/main/kotlin/jp/ergo/android/screentouchdispatcher/ScreenTouchDispatcherBroadcastReceiver.kt
package jp.ergo.android.screentouchdispatcher

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.preference.PreferenceManager
import android.widget.Toast
import jp.ergo.android.screentouchdispatcher.util.Logger

class ScreenTouchDispatcherBroadcastReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        Logger.d(TAG, "onReceive() " + intent.action)
        Toast.makeText(context, "受け取ったよ", Toast.LENGTH_LONG).show()

        val action = intent.action
        val serviceIntent = Intent(context, ScreenTouchDispatcherService::class.java)
        if (action == ACTION_SWITCH) {
            val isDispatching = isDispatching(context)
            Logger.d(TAG, "isDispatching: " + isDispatching)
            setDispatching(context, !isDispatching)
            if (isDispatching) {
                context.stopService(serviceIntent)
                NotificationUtils.sendNotification(context, "タッチして画面をロック")
            } else {
                context.startService(serviceIntent)
                NotificationUtils.sendNotification(context, "タッチしてロックを解除")
            }
        }
    }


    private fun isDispatching(context: Context): Boolean {
        return PreferenceManager.getDefaultSharedPreferences(context).getBoolean("isDispatching", false)
    }

    private fun setDispatching(context: Context, isDispatching: Boolean) {
        PreferenceManager.getDefaultSharedPreferences(context).edit().putBoolean("isDispatching", isDispatching).commit()

    }

    companion object {

        private val TAG = ScreenTouchDispatcherBroadcastReceiver::class.java.simpleName

        val ACTION_START_SERVICE = "jp.ergo.android.screentouchdispatcher.SATRT_SERVICE"
        val ACTION_STOP_SERVICE = "jp.ergo.android.screentouchdispatcher.STOP_SERVICE"
        val ACTION_SWITCH = "jp.ergo.android.screentouchdispatcher.SWITCH"
    }

}